runtimes/typescript/lbr-prelude/src/LambdaBuffers/Runtime/AvlTree.ts

```diff
@@ -0,0 +1,270 @@
+import type { Ord } from "./Ord.js";
+
+/**
+ * `Node` is an AVL tree i.e., a height balanced tree.
+ *
+ * Recall that an AVL tree satisfies the invariants:
+ *
+ * - the absolute value of the difference of height between `left` and `right` is
+ *   at most `1`.
+ */
+export type Node<K> =
+  | null
+  | { element: K; height: number; left: Node<K>; right: Node<K> };
+
```

Comment:
This is killing me, we can't maintain this whole lot.

Is there no libraries we can reuse that seem stable? https://github.com/AlexAegis/avl/tree/master looks like smtn, but I would hope that there's the defacto standard containters lib widely used